### PR TITLE
[DOCKER] Use postgres as back end for hive metastore

### PR DESCRIPTION
**Pull request by honzasterba**
hopefully this will make hadoop tests more stable

### CODE REVIEW

File: docker/hadoop/common/startup/51_postgres_init [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#!/bin/bash -ex
+
+# init database for metastore
+
+#rm -rf /var/lib/mysql
```
Comment (michal-raska) [inline docker/hadoop/common/startup/51_postgres_init]: Can we remove the commented lines please?

File: docker/hadoop/cdh/Dockerfile [hunk 2]
```diff
@@ -89,11 +89,9 @@ RUN chmod 700 /usr/sbin/startup.sh && \
     chown -R hive:hive /opt/hive-scripts && \
     chmod +x /usr/sbin/install_hive.sh && \
     chmod +x /usr/sbin/install_ldap.sh && \
-    chmod +x /usr/sbin/mysql_configure.sh && \
     sync && \
     /usr/sbin/install_hive.sh && \
-    /usr/sbin/mysql_configure.sh && \
-    ln -sf /usr/share/java/mysql-connector-java.jar /usr/lib/hive/lib/mysql-connector-java.jar
+    ln -sf /usr/share/java/postgresql-jdbc4.jar /usr/lib/hive/lib/postgresql-jdbc4.jar
```
Comment (michal-raska) [inline docker/hadoop/cdh/Dockerfile]: There's a similar linking in `docker/hadoop/hdp/scripts/sbin/install_hive.sh`. I would like if they will be handled in one place, but that's probably out of scope of this PR.

Comment (honzasterba) [reply docker/hadoop/cdh/Dockerfile]: yes, the difference is that HIVE_HOME is not needed here

File: docker/hadoop/common/startup/51_postgres_init (hunk 1 above)
Comment (honzasterba) [reply docker/hadoop/common/startup/51_postgres_init]: good catch, forgot about those

**Pull request merged**